nullaway/src/main/java/com/uber/nullaway/handlers/InferredJARModelsHandler.java

```diff
@@ -0,0 +1,224 @@
+
+      if (!argAnnotCache.containsKey(className)) {
+        if (VERBOSE) {
+          System.out.println("[JI Warn] Cannnot find Annotation Cache for class " + className);
+        }
+        return nonNullPositions;
+      }
+      Map<Integer, Set<String>> methodArgAnnotations = argAnnotCache.get(className).get(methodSign);
+      if (methodArgAnnotations == null) {
+        if (VERBOSE) {
+          System.out.println(
+              "[JI Warn] Cannnot find Annotation Cache entry for method "
+                  + methodSign
+                  + " in "
+                  + className);
+        }
+        return nonNullPositions;
+      }
+      if (DEBUG) {
+        System.out.println(
+            "[JI Debug] Found Annotation Cache entry for method "
+                + methodSign
+                + " in "
+                + className
+                + " -- "
+                + methodArgAnnotations.toString());
+      }
+      Set<Integer> jiNonNullParams = new LinkedHashSet<>();
+      for (Map.Entry<Integer, Set<String>> annotationEntry : methodArgAnnotations.entrySet()) {
+        if (annotationEntry.getValue().contains("javax.annotation.Nonnull")) {
+          // Parameters are 0-indexed, vs. 1-indexed in WALA
+          jiNonNullParams.add(annotationEntry.getKey() - 1);
+        }
+      }
+      if (DEBUG) {
+        System.out.println("[JI DEBUG] Nonnull params  : " + jiNonNullParams.toString());
+      }
+      return Sets.union(nonNullPositions, jiNonNullParams).immutableCopy();
+    } catch (ClassNotFoundException cnfe) {
+      if (VERBOSE) {
+        System.out.println("[JI Warn] Cannnot find Class " + className);
+      }
+      return nonNullPositions;
+    } catch (IOException e) {
+      throw new Error(e);
+    }
+  }
+
+  private void parseStubStream(
+      InputStream stubxInputStream,
+      String stubxLocation,
+      Map<String, Map<Integer, Set<String>>> argAnnotations)
+      throws IOException {
+    String[] strings;
+    DataInputStream in = new DataInputStream(stubxInputStream);
+    // Read and check the magic version number
+    if (in.readInt() != VERSION_0_FILE_MAGIC_NUMBER) {
+      throw new Error("Invalid file version/magic number for stubx file!" + stubxLocation);
+    }
+    // Read the number of strings in the string dictionary
+    int numStrings = in.readInt();
+    // Populate the string dictionary {idx => value}, where idx is encoded by the string position
+    // inside this section.
+    strings = new String[numStrings];
+    for (int i = 0; i < numStrings; ++i) {
+      strings[i] = in.readUTF();
+    }
+    // Read the number of (package, annotation) entries
+    int numPackages = in.readInt();
+    // Read each (package, annotation) entry, where the int values point into the string
+    // dictionary loaded before.
+    for (int i = 0; i < numPackages; ++i) {
+      String packageName = strings[in.readInt()];
+      String annotation = strings[in.readInt()];
```

Comment:
Nit: You could just have something like 

`in.readInt(); in.readInt();  // Not using this info for now`

instead of saving to unused variables, and it might be slightly clearer.